File: spec/java/src/io/kaitai/struct/spec/TestExpr1.java
package io.kaitai.struct.spec;

import io.kaitai.struct.testformats.Expr1;
import org.testng.annotations.Test;

import static org.testng.Assert.assertEquals;

public class TestExpr1 extends CommonSpec {
    @Test
    public void testExpr1() throws Exception {
        Expr1 r = Expr1.fromFile(SRC_DIR + "str_encodings.bin");

        assertEquals(r.lenOf1(), 10);
        assertEquals(r.lenOf1Mod().intValue(), 8);
        assertEquals(r.str1(), "Some ASC");
        assertEquals(r.str1Len().intValue(), 8);
    }
}

File: spec/rust/tests/test_expr_1.rs
#![allow(dead_code)]
use std::fs;

extern crate kaitai;
use self::kaitai::*;
mod formats;
use formats::expr_1::*;

#[test]
fn test_expr_1() {
    let bytes = fs::read("../../src/str_encodings.bin").unwrap();
    let reader = BytesReader::new(&bytes);
    let mut r = Expr1::default();

    if let Err(err) = r.read(&reader, None, KStructUnit::parent_stack()) {

        panic!("{:?}", err);
    }
    assert_eq!(10, r.len_of_1);

    let res = r.len_of_1_mod(&reader);
    assert_eq!(8, *res.unwrap());

    assert_eq!("Some ASC", *r.str1());

    let res = r.str1_len(&reader);
    assert_eq!(8, *res.unwrap());
}
